QuickFIXn/Fields/FieldBase.cs

```diff
@@ -98,13 +98,20 @@ public int GetTotal()
         if (_changed)
             MakeStringFields();
 
-        int sum = 0;
-        byte[] array = CharEncoding.GetBytes(_stringField);
-        foreach (byte b in array)
+        ValueDisposable disposable = CharEncoding.GetBytes(_stringField, out ReadOnlySpan<byte> array);
```

Comment:
With the ref struct suggestion this will work:

```suggestion
        using ValueDisposable disposable = CharEncoding.GetBytes(_stringField, out ReadOnlySpan<byte> array);
```

Reply:
> With the ref struct suggestion this will work:

I am aware that the using syntax works as expected. However, I noticed that it generates some additional overhead in the IL code. The final compiled output is as follows:
```csharp
ValueDisposable disposable = CharEncoding.GetBytes(_stringField, out ReadOnlySpan<byte> array);
try
{
    ...
}
finally
{
    if (disposable != null)
    {
        disposable.Dispose();
    }
}
```

Reply:
IMO that would be acceptable for a more concise usage pattern

Reply:
Does `using` eliminate the need to explicitly call Dispose?

Reply:
I've changed it to not call `using` and `try-catch`.